docs/guides/device-gateway/sdk.md

```diff
@@ -0,0 +1,344 @@
+
+![alt-text](img/authtoken.png)
+
+## Configure the sample agent
+
+The sample application uses the ngrok Python SDK to embed connectivity directly into the application
+rather than using the standalone ngrok agent, the CLI, or the ngrok Operator. It serves a local REST
+API you can use to create, delete, and list tunnels.
+
+### Clone the repo
+
+Clone [this sample application](https://github.com/ngrok-samples/ngrok-iot-agent) on the device or device gateway you wish to connect to.
+
+### Create and activate a virtual environment
+
+Run the following to create a virtual environment, providing a directory for `{PATH_TO_VENV}`:
+
+`python -m venv {PATH_TO_VENV}`
+
+Run the appropriate command for your O/S to activate the virtual environment:
+
+- In cmd.exe
+  `{PATH_TO_VENV}\Scripts\activate.bat`
+- In PowerShell
+  `{PATH_TO_VENV}\Scripts\Activate.ps1`
+- Linux and MacOS
+  `source {PATH_TO_VENV}/bin/activate`
+
+### Install dependencies
+
+Run `pip install -r requirements` to install the dependencies.
+
+### Update the domain
+
+Update line `109` with your domain. Since you created a wildcard domain, you can use any subdomain
+of your wildcard domain without reserving it. For example, for the wildcard domain `*.sitea.configurable-domain.com`, you could use `api.sitea.configurable-domain.com`.
+
+### Set your authtoken
+
+This sample application is designed to read your ngrok credentials from the environment, so run the
+following command, replacing `{YOUR_AUTHTOKEN}` with the authtoken you created in the previous
+step:
+
+`export NGROK_AUTHTOKEN={YOUR_AUTHTOKEN}`
+
+## Use the sample app
+
+This sample application is a custom ngrok agent built using the Python SDK. In a previous step,
+you updated the domain on line `109` of `agent.py`.
+
+This domain will be used to serve a local REST API built into the sample application using FastAPI.
+You'll use this API to create, delete, and list tunnels remotely.
+
+The application persists tunnels in a local JSON file, `ngrok_config.json`, and recreates them when the application is restarted. You could extend this application to store tunnel configuration in a backend database.
+
+You may pass policy information directly in the body of the request. The application includes a JSON
+file, `test.json`, which contains the default [policies](https://ngrok.com/docs/http/traffic-policy/) it applies to any new request that doesn't specify policy information. You could update this file to control policies applied to new tunnels rather than passing the policy information in each request.
+
+### Start the app
+
+Run the following to start the agent:
+
+```python
+python agent.py
+```
+
+### List tunnels
+
+Run the following command to list tunnels started by this custom agent application,
+substituting the domain you set on line `109` for `{YOUR_AGENT_DOMAIN}`:
+
+```curl
+curl -k \
+-X GET \
+https://{YOUR_AGENT_DOMAIN}/tunnels
+```
```

Comment:
````suggestion
```curl
curl -k \
  -X GET \
  https://{YOUR_AGENT_DOMAIN}/tunnels
```
````

I like to format my multiline commands a little... same vibe as before, just a thought for readability!